econplayground/main/tests/test_views.py

```diff
@@ -174,6 +174,77 @@ def test_edit_cohort(self):
         self.assertFalse('messages' in response.cookies)
 
 
+class CohortDeleteViewTest(LoggedInTestInstructorMixin, TestCase):
+    def setUp(self):
```

Comment:
Deleting a Cohort that contains Topics and associated Graphs raises an exception: "Cannot delete some instances of model 'Topic' because they are referenced through a protected foreign key: 'Graph.topic'"

Reply:
I've added a failing test for this, thanks. I'll now sort out how to handle this...

Reply:
I've decided to preserve the graphs on course delete - the Graph's 'topic' attribute is set to NULL in this case.